webui/src/pages/user/user-namespace-extension-list.tsx

```diff
@@ -0,0 +1,107 @@
+/********************************************************************************
+ * Copyright (c) 2020 TypeFox and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { Theme, createStyles, WithStyles, withStyles, Typography } from '@material-ui/core';
+import { Namespace, isError, Extension, ErrorResult } from '../../extension-registry-types';
+import { UserNamespaceExtensionListItem } from './user-namespace-extension-list-item';
+import { ExtensionRegistryService } from '../../extension-registry-service';
+import { ErrorResponse } from '../../server-request';
+import { DelayedLoadIndicator } from '../../custom-mui-components/delayed-load-indicator';
+
+const extensionListStyles = (theme: Theme) => createStyles({
+    container: {
+        padding: theme.spacing(5, 0),
+    },
+    extensions: {
+        display: 'grid',
+        gridTemplateColumns: `repeat(auto-fit, minmax(200px, 1fr))`,
+        gap: `1.5rem`,
+        marginTop: '1rem',
+    }
+});
+
+
+class UserNamespaceExtensionListComponent extends React.Component<UserNamespaceExtensionListComponent.Props, UserNamespaceExtensionListComponent.State> {
+
+    constructor(props: UserNamespaceExtensionListComponent.Props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            extensions: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.initExtensions();
+    }
```

Comment:
You also need a `componentDidUpdate(...)` method to update the extensions when the user selects another namespace. Currently the same extensions are shown after switching namespace.

Reply:
Sorry! I don't know React well. Isn't it that when props or state for a component are changed it gets re-rendered?

The extensions that are rendered are the ones from the `choosenNamespace` which is a piece of state. I was expecting the state `choosenNamespace` to be changed when another namespace is selected via the Tab(maybe that is not the case).

![image](https://user-images.githubusercontent.com/46004116/81839198-a8d2ae80-9560-11ea-9e5e-fba1a9a4e02d.png)

Please explain If I misunderstood the implementation details?

Reply:
The namespace is in the state of the parent component, but in the props of this component. You are reading that namespace after the component has mount, but it is not re-mounted when the props changed. You need to add `componentDidUpdate(...)`, compare the previous props with the new props to check whether the namespace has changed, and in that case call `initExtensions()` again (could rename that to `loadExtensions()`).

You can see this pattern in other components of this project, just search for `componentDidUpdate`.